app/Filament/Resources/User/UserResource/Pages/EditUser.php

```diff
@@ -51,42 +51,57 @@ protected function beforeSave(): void
             return;
         }
 
-        $rconEnabled = config('hotel.rcon.enabled');
-        $rcon = app(RconService::class);
+        $rconService = app(RconService::class);
 
         if (!$user->online) {
-            $this->treatChangedCurrenciesWithoutRcon($user, $data);
+            $this->updateOfflineUserWithoutRcon($user, $data);
             return;
         }
 
-        if ($user->online && !$rconEnabled) {
+        if ($user->online && !$rconService->isConnected()) {
             Notification::make()
                 ->danger()
-                ->title(__('RCON is not enabled!'))
-                ->body(__('You cannot edit users because RCON is not enabled and the user is online.'))
+                ->title(__('RCON is not connected!'))
+                ->body(__('You cannot edit users because RCON is not connected and the user is online.'))
                 ->send();
 
             $this->halt();
             return;
         }
 
-        if ($data['credits'] != $user->credits) {
-            $rcon->sendSafelyFromDashboard('giveCurrency',
-                [$user, 'credits', -$user->credits + $data['credits']],
-                'RCON: Failed to send credits'
-            );
-        }
-
-        $this->checkUsernameChangedPermission($user, $data, $rcon);
-        $this->treatChangedCurrencies($user, $data, $rcon);
-        $this->treatChangedUserRank($user, $data, $rcon);
-        $this->treatChangedUserMotto($user, $data, $rcon);
+        $this->updateUserMotto($user, $data, $rconService);
+        $this->updateUserCredits($user, $data, $rconService);
+        $this->updateUserCurrencies($user, $data, $rconService);
+        $this->checkUsernameChangePermission($user, $data, $rconService);
+        $this->updateUserRank($user, $data, $rconService);
     }
 
-    private function treatChangedCurrenciesWithoutRcon(Model $user, array $data): void
+    private function updateOfflineUserWithoutRcon(Model $user, array $data): void
     {
+        if ($data['motto'] !== $user->motto) {
+            $user->update([
+                'motto' => $data['motto'] != '' ? $data['motto'] : '',
+            ]);
+
+            Notification::make()
+                ->success()
+                ->title(__('Success!'))
+                ->body(__('Successfully updated user motto.'))
+                ->send();
+        }
+
         if ($data['credits'] != $user->credits) {
-            $user->credits = $data['credits'];
+            $total = (0 - $user->credits) + $data['credits'];
```

Comment:
Is this correct? As to my understanding this will not update the users credit amount to be of the specified, instead it will simply "boost" their balance ever so slightly.

If this is meant as a way to simply "add" X amount to the current balance, then please respond letting me know so, as in such case it is correctly done :D